pkg/trace/writer/payload.go

```diff
@@ -140,6 +141,9 @@ func (s *queuableSender) doSend(payload *payload) (sendStats, error) {
 	}
 
 	startFlush := time.Now()
+	defer func(start time.Time) {
```

Comment:
Nit: do we need to do this passing of `startFlush` as an argument to this anonymous function? It's slightly simpler to just reference the variable directly: `metrics.Timing("datadog.trace_agent.debug.send_payload_ms", time.Since(startFlush), nil, 1)`

Same goes for all the other `metrics.Timing` calls we're doing.

Reply:
In this case we could simply use it without having a function argument, that's true. In all other cases I prefer the current form. It makes it clear what function scope the time variable belongs too. As it has no use outside the defer function, I'd prefer it wouldn't exist outside.